xwiki-platform-core/xwiki-platform-ckeditor/xwiki-platform-ckeditor-plugins/src/main/resources/xwiki-focusedplaceholder/plugin.js

```diff
@@ -0,0 +1,268 @@
+            for (i = 0; i < element.childNodes.length; i++) {
+                var child = element.childNodes[i];
+                // Child is not empty if it doesn't have a nodeName
+                if (child.nodeName === undefined) {
+                    return false;
+                }
+
+                // Child is not considered empty if it is not ignored if empty
+                if (!this.editor.config.focusedplaceholder.ignoreIfEmpty.includes(child.nodeName.toLowerCase())) {
+                    return false;
+                }
+
+                // Child is not considered empty if it is not empty
+                if (!this.isEmpty(child)) {
+                    return false;
+                }
+            }
+
+            return true;
+        },
+
+        /**
+         * Finds the first configured ancestor of a DOM element
+         *
+         * @param {Object} element
+         * @return {Object} - The first configured ancestor
+         */
+        getFirstConfiguredAncestor: function(element) {
+
+            // No parents
+            if (element === null) {
+                return element;
+            }
+            if (element.parentNode === null) {
+                return element.parentNode;
+            }
+
+            // Configured element
+            if (element.tagName !== undefined &&
+                this.editor.config.focusedplaceholder.placeholder[element.tagName.toLowerCase()] !== undefined) {
+                return element;
+            }
+
+            return this.getFirstConfiguredAncestor(element.parentNode);
+
+        },
+
+        /**
+         * Updates the placeholder when the caret moves
+         *
+         * @param {CKEDITOR.dom.range}
+         */
+        updatePlaceholder: function(range) {
+            // Clear previous placeholders
+            this.removeAllPlaceholders(range.document.$);
+
+            // No placeholder when a selection is occuring
+            if (!range.collapsed) {
+                return null;
+            }
+
+            // Find the matching ancestor
+            var ancestor = this.getFirstConfiguredAncestor(range.getCommonAncestor().$);
+            if (ancestor === null) {
+                ancestor = range.getCommonAncestor().$;
+            }
+
+            // Add placeholder when ancestor is considered empty
+            if (this.isEmpty(ancestor)) {
+                ancestor.setAttribute(ATTRIBUTE_NAME, this.getPlaceholderContent(ancestor.tagName.toLowerCase()));
+            }
+        },
+
+        /**
+         * Returns the expected placeholder text for a given element
+         *
+         * @param {string} tag - Lowercase tagName
+         * @return {string}
+         */
+        getPlaceholderContent: function(tag) {
+
+            var translationKey = this.editor.config.focusedplaceholder.placeholder[tag];
+
+            if (translationKey === null) {
+                translationKey = this.editor.config.focusedplaceholder.placeholder._fallback;
```

Comment:
The fallback value is currently empty...

Reply:
Thank you, the fallback was indeed not used.